src/index.js

```diff
@@ -1,35 +1,97 @@
 // 1. Add function code goes here
-exports.add = function add() {};
+exports.add = function add(x, y) {
+  return x + y
+};
 
 // 2. Multiply function code goes here
-exports.multiply = function multiply() {};
+exports.multiply = function multiply(x, y) {
+  return x * y
+};
 
 // 3. OddOrEven function code goes here
-exports.oddOrEven = function oddOrEven() {};
+exports.oddOrEven = function oddOrEven(x) {
+  if(x % 2 == 0) {
+    return ('even')
+  } else {
+    return ('odd')
+  }
+};
 
 // 4. Write a function that returns an array that includes number 1 to 100
 // Ex: [1,2,3,4, ..., 99, 100]
-exports.arrayGenerator = function arrayGenerator() {};
+exports.arrayGenerator = function arrayGenerator() {
+  let numbers = [];
+  for(i=1; i<=100; i++) {
+    numbers.push(i);
+  }
+  return(numbers);
+};
 
 // 5. Fix this function. We want to see 2 in the console instead of undefined
 exports.hoisting = function hoisting() {
-  console.log(y); // undefined
   let y = 2;
+  return (y);
 };
 
 // 6. Write a function that accepts unlimited amount of numbers as input
 // and return the smallest value
-exports.minValue = function minValue() {};
+exports.minValue = function minValue(...numbers) {
+  return Math.min(...numbers);
+  };
 
 // 7. Write a function that accepts an array of numbers as input
 // and return a new array with all numbers doubled
 // Ex: [1,2,3] => [2,4,6]
-exports.doubleArray = function doubleArray() {};
+exports.doubleArray = function doubleArray([...x]) {
+  let doubledArray = [];
+  for(i = 0; i < x.length; i++) {
+    doubledArray.push(x[i] * 2); 
+  };
```

Comment:
Let's use higher order function instead, check out the map method of array in javascript

Reply:
Thanks for the feedback, I updated the function!